privaterelay/tests/utils_tests.py

```diff
@@ -20,3 +37,190 @@ def test_get_premium_country_lang(self):
+        assert isinstance(user, User)
+        return user
+    return None
+
+
+def test_flag_is_active_for_task_missing_flag(
+    flag_user: User | None, caplog: LogCaptureFixture
+) -> None:
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    assert not Flag.objects.filter(name=TEST_FLAG_NAME).exists()
+
+
+def test_flag_is_active_for_task_missing_flag_logged(
+    flag_user: User | None, waffle_settings: SettingsWrapper, caplog: LogCaptureFixture
+) -> None:
+    waffle_settings.WAFFLE_LOG_MISSING_FLAGS = logging.WARNING
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == [
+        ("waffle", logging.WARNING, f"Flag {TEST_FLAG_NAME} not found")
+    ]
+    assert not Flag.objects.filter(name=TEST_FLAG_NAME).exists()
+
+
+def test_flag_is_active_for_task_missing_flag_created(
+    flag_user: User | None,
+    waffle_settings: SettingsWrapper,
+    waffle_cache: BaseCache,
+    caplog: LogCaptureFixture,
+) -> None:
+    waffle_settings.WAFFLE_CREATE_MISSING_FLAGS = True
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    flag = Flag.objects.get(name=TEST_FLAG_NAME)
+    assert waffle_cache.get(flag._cache_key(TEST_FLAG_NAME)) == flag
+
+
+def test_flag_is_active_for_task_missing_flag_created_with_default_true(
+    flag_user: User | None,
+    waffle_settings: SettingsWrapper,
+    waffle_cache: BaseCache,
+    caplog: LogCaptureFixture,
+) -> None:
+    waffle_settings.WAFFLE_CREATE_MISSING_FLAGS = True
+    waffle_settings.WAFFLE_FLAG_DEFAULT = True
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    flag = Flag.objects.get(name=TEST_FLAG_NAME)
+    assert waffle_cache.get(flag._cache_key(TEST_FLAG_NAME)) == flag
+
+
+def test_flag_is_active_for_task_missing_flag_with_default_true(
+    flag_user: User | None, waffle_settings: SettingsWrapper, caplog: LogCaptureFixture
+) -> None:
+    waffle_settings.WAFFLE_FLAG_DEFAULT = True
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+    assert caplog.record_tuples == []
+    assert not Flag.objects.filter(name=TEST_FLAG_NAME).exists()
+
+
+def test_flag_is_active_for_task_existing_but_off_flag(flag_user: User | None) -> None:
+    flag = Flag.objects.create(name=TEST_FLAG_NAME)
+    assert flag.everyone is None
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_with_everyone_on_flag(flag_user: User | None) -> None:
+    Flag.objects.create(name=TEST_FLAG_NAME, everyone=True)
+    assert flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
+def test_flag_is_active_for_task_with_everyone_off_flag(flag_user: User | None) -> None:
+    Flag.objects.create(name=TEST_FLAG_NAME, everyone=False)
+    assert not flag_is_active_in_task(TEST_FLAG_NAME, flag_user)
+
+
```

Comment:
Suggestion (blocking) add a test for `everyone=False`, `user=True`, and confirm `flag_is_active_in_task` returns `False`